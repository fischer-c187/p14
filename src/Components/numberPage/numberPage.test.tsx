import { render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import userEvent from "@testing-library/user-event";
import NumberPage from "./numberPage";

describe("NumberPage", () => {
  it("should render correctly", () => {
    render(<NumberPage currentPage={5} totalPage={10} />);
    expect(screen.getAllByTestId("pageenabled").length).toEqual(7);
    expect(screen.getAllByTestId("pagedisabled").length).toEqual(2);
  });

  it("should call onClick on click", async () => {
    const onClick = vi.fn();
    const user = userEvent.setup();
    render(<NumberPage currentPage={5} totalPage={10} onClick={onClick} />);

    await user.click(screen.getByText("6"));
    expect(onClick).toHaveBeenCalled();
  });

  it("should render expected snapshot", () => {
    const { asFragment } = render(
      <NumberPage currentPage={5} totalPage={10} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render correctly with siblingCount", () => {
    render(<NumberPage currentPage={5} totalPage={10} siblingCount={1} />);
    expect(screen.getByText("4")).toBeInTheDocument();
    expect(screen.getByText("6")).toBeInTheDocument();
    expect(screen.queryByText("7")).not.toBeInTheDocument();
    expect(screen.queryByText("3")).not.toBeInTheDocument();
  });
});
