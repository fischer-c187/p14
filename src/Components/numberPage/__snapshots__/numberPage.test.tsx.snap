// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NumberPage > should render expected snapshot 1`] = `
<DocumentFragment>
  <ul
    class="flex justify-center items-center gap-1"
  >
    <li>
      <button
        class="p-3 text-sm text-gray-500 font-medium rounded-lg w-10 h-10 leading-none transition-all hover:bg-gray-200"
        data-pagenumber="1"
        data-testid="pageenabled"
        type="button"
      >
        1
      </button>
    </li>
    <li>
      <button
        class="p-3 text-sm text-gray-500 font-medium rounded-lg w-10 h-10 leading-none transition-all cursor-default"
        data-pagenumber="..."
        data-testid="pagedisabled"
        disabled=""
        type="button"
      >
        ...
      </button>
    </li>
    <li>
      <button
        class="p-3 text-sm text-gray-500 font-medium rounded-lg w-10 h-10 leading-none transition-all hover:bg-gray-200"
        data-pagenumber="3"
        data-testid="pageenabled"
        type="button"
      >
        3
      </button>
    </li>
    <li>
      <button
        class="p-3 text-sm text-gray-500 font-medium rounded-lg w-10 h-10 leading-none transition-all hover:bg-gray-200"
        data-pagenumber="4"
        data-testid="pageenabled"
        type="button"
      >
        4
      </button>
    </li>
    <li>
      <button
        class="p-3 text-sm font-medium rounded-lg w-10 h-10 leading-none transition-all bg-citron-600 text-gray-100 hover:bg-citron-600"
        data-pagenumber="5"
        data-testid="pageenabled"
        type="button"
      >
        5
      </button>
    </li>
    <li>
      <button
        class="p-3 text-sm text-gray-500 font-medium rounded-lg w-10 h-10 leading-none transition-all hover:bg-gray-200"
        data-pagenumber="6"
        data-testid="pageenabled"
        type="button"
      >
        6
      </button>
    </li>
    <li>
      <button
        class="p-3 text-sm text-gray-500 font-medium rounded-lg w-10 h-10 leading-none transition-all hover:bg-gray-200"
        data-pagenumber="7"
        data-testid="pageenabled"
        type="button"
      >
        7
      </button>
    </li>
    <li>
      <button
        class="p-3 text-sm text-gray-500 font-medium rounded-lg w-10 h-10 leading-none transition-all cursor-default"
        data-pagenumber="..."
        data-testid="pagedisabled"
        disabled=""
        type="button"
      >
        ...
      </button>
    </li>
    <li>
      <button
        class="p-3 text-sm text-gray-500 font-medium rounded-lg w-10 h-10 leading-none transition-all hover:bg-gray-200"
        data-pagenumber="10"
        data-testid="pageenabled"
        type="button"
      >
        10
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
